datadog_checks_base/tox.ini

```diff
@@ -14,10 +14,11 @@ dd_check_style = true
 dd_check_types = true
 dd_mypy_args =
     --py2
-    --disallow-untyped-defs
+    --check-untyped-defs
```

Comment:
Allows us to have mypy run on functions even if they don't have a signature.

I'm thinking this is something we might want to put into our root `mypy.ini`, since it _can_ be cumbersome to have to specify function signatures even when they're irrelevant (e.g. a bunch of `Any`) just to have mypy check the inside of the function. This more lenient behavior is probably better suited for our workflow.